File: Day13/5_Check_For_Balanced_Parantheses.java
class stack{
    char arr[] = new char[10000];
    int top;
    public stack(){
        this.top = -1;
    }

    public void push(char x){
        this.arr[++this.top] = x;
    }

    public char pop() {
        if (top == -1)
            return '1';

        return this.arr[this.top--];
    }
}

class Solution {
    public boolean isValid(String s) {
        stack st = new stack();
        for (char ch : s.toCharArray()){
            if (ch == '(' || ch == '{' || ch == '[') {
                st.push(ch);
            } else {
                char v = st.pop();
                if (v == '(' && ch == ')') {
                    continue;
                } else if (v == '{' && ch == '}') {
                    continue;
                } else if (v == '[' && ch == ']') {
                    continue;
                } else {
                    return false;
                }
            }
        }

        if (st.pop() == '1')
            return true;
        else
            return false;
    }
}

File: Day13/5_Check_For_Balanced_Parantheses.py
class stack:
    def __init__(self):
        self.arr = []

    def push(self,x):
        self.arr.append(x)

    def pop(self):
        if len(self.arr) == 0:
            return -1
        return self.arr.pop()

class Solution:
    def isValid(self, s: str) -> bool:
        Stack = stack()
        for i in s:
            if i == '(' or i == '[' or i == '{':
                Stack.push(i)
            else:
                v = Stack.pop()
                if v == '(' and i == ')':
                    continue
                elif v == '[' and i == ']':
                    continue
                elif v == '{' and i == '}':
                    continue
                else:
                    return False

        if Stack.pop() == -1:
            return True
        else:
            return False
